### PR TITLE
Standalone PageRank

### PR DESCRIPTION
**Pull request by ankurdave**
An implementation of PageRank directly on the GraphX primitives.

### CODE REVIEW

File: graph/src/main/scala/org/apache/spark/graph/algorithms/PageRank.scala [hunk 1]
```diff
@@ -151,4 +152,57 @@ object PageRank {
     Pregel(pagerankGraph, initialMessage)(vertexProgram, sendMessage, messageCombiner)
       .mapVertices((vid, attr) => attr._1)
   } // end of deltaPageRank
+
+  def runStandalone[VD: Manifest, ED: Manifest](
+      graph: Graph[VD, ED], tol: Double, resetProb: Double = 0.15): VertexRDD[Double] = {
+
+    // Initialize the ranks
+    var ranks: VertexRDD[Double] = graph.vertices.mapValues((vid, attr) => 0.0).cache()
```
Comment (jegonzal) [inline graph/src/main/scala/org/apache/spark/graph/algorithms/PageRank.scala]: The zero vector is a degenerate eigenvector.  You want to start everything with rank resetProb.

Comment (ankurdave) [inline graph/src/main/scala/org/apache/spark/graph/algorithms/PageRank.scala]: Done. I tried that earlier, but the ranks still diverge and grow to infinity.

**Comment by ankurdave** [comment]
@jegonzal @rxin I fixed the rank-delta confusion, but there's still a bug that appears in the second iteration. For a 2x2 grid graph, the ranks should be

```
0, 0.150000
1, 0.213750
2, 0.213750
3, 0.513375
```

Instead, they are

```
0, 0.150000
1, 0.213750
2, 0.213750
3, 0.532500
```

**Comment by ankurdave** [comment]
This now passes tests and is ready to merge, pending a Jenkins run.

**Pull request merged**